### PR TITLE
[ledger-api-test-tool] - Add ability to disable tests based on ledger features [kvl-1218]

### PR DESCRIPTION
**Pull request by nicu-da**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala [hunk 1]
```diff
@@ -75,12 +75,17 @@ final class LedgerTestCasesRunner(
...
         .createTestContext(testName, identifierSuffix)
         .flatMap { context =>
-          val start = System.nanoTime()
-          val result = test(context).map(_ => Duration.fromNanos(System.nanoTime() - start))
-          logger.info(
-            s"Started '${test.description}'${test.repetition.fold("")(r => s" (${r._1}/${r._2})")} with a timeout of $scaledTimeout."
-          )
-          result
+          val features = context.configuredParticipants.head.features
+          if (test.testCase.enabled(features)) {
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala:79]: We currently manage other exclusions much higher up; is it possible to move this logic to the runner?

**Review by SamirTalwar-DA — APPROVED** [review]
Seems legitimate. One question.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala [hunk 2]
```diff
@@ -75,12 +75,17 @@ final class LedgerTestCasesRunner(
       session
         .createTestContext(testName, identifierSuffix)
         .flatMap { context =>
-          val start = System.nanoTime()
-          val result = test(context).map(_ => Duration.fromNanos(System.nanoTime() - start))
-          logger.info(
-            s"Started '${test.description}'${test.repetition.fold("")(r => s" (${r._1}/${r._2})")} with a timeout of $scaledTimeout."
-          )
-          result
+          val features = context.configuredParticipants.head.features
```
Comment (hubert-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala:78]: Huh, I thought there has already been support for that here.

**Review by hubert-da — APPROVED** [review]
👍🏻

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala (hunk 1 above)
Comment (nicu-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala]: Moved it to the highest level where possible. We have one higher level, but we don't have access to the features there.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala (hunk 2 above)
Comment (nicu-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala]: Unfortunately not, the features are associated to a session/participant so we have to just choose one to get the features

**Review by SamirTalwar-DA — APPROVED** [review]

**Review by hubert-da — APPROVED** [review]
👍🏻 I'm just wondering if this can somehow interfere with the exclusion mechanism (`--exclude`) by a misleading output, for example.

**Comment by nicu-da** [comment]
> 👍🏻 I'm just wondering if this can somehow interfere with the exclusion mechanism (`--exclude`) by a misleading output, for example.

Not really. 
The `--exclude` mechanism is accounted for before this check. The tests which are excluded through that mechanism will not get into the runner. 
The tests excluded through the `Result.Excluded` exception will actually be reported in the success section. But they will be printed in yellow with the exclusion reason.

**Review by mziolekda — APPROVED** [review]
Nice improvement!

**Pull request merged**